packages/hash/frontend/src/pages/[account-slug]/types/entity-type/property-type-utils.ts

```diff
@@ -0,0 +1,133 @@
+import { PropertyValues, VersionedUri } from "@blockprotocol/type-system-web";
+import { faList } from "@fortawesome/free-solid-svg-icons";
+import { types } from "@hashintel/hash-shared/types";
+import { fa100 } from "../../../../shared/icons/pro/fa-100";
+import { faCube } from "../../../../shared/icons/pro/fa-cube";
+import { faSquareCheck } from "../../../../shared/icons/pro/fa-square-check";
+import { faText } from "../../../../shared/icons/pro/fa-text";
+
+export type ExpectedValue =
+  | VersionedUri
+  | {
+      typeId: "array";
+      id: string;
+      flattenedProperties: Record<string, DataType>;
+    };
+export interface PrimitiveTypeData {
+  typeId: VersionedUri;
+}
+
+export interface ArrayTypeData {
+  typeId: "array";
+  minItems: number;
+  maxItems?: number;
+  expectedValues: string[];
+}
+
+export interface DataType {
+  id: string;
+  parentId?: string;
+  data?: PrimitiveTypeData | ArrayTypeData;
+}
+
+export const dataTypeOptions = [
+  types.dataType.text.dataTypeId,
+  types.dataType.number.dataTypeId,
+  types.dataType.boolean.dataTypeId,
+];
+
+export const customDataTypeOptions = [
+  "array",
+  types.dataType.object.dataTypeId,
+];
+
+export const dataTypeData = {
+  [types.dataType.text.dataTypeId]: {
+    title: types.dataType.text.title,
+    icon: faText,
+  },
+  [types.dataType.number.dataTypeId]: {
+    title: types.dataType.number.title,
+    icon: fa100,
+  },
+  [types.dataType.boolean.dataTypeId]: {
+    title: types.dataType.boolean.title,
+    icon: faSquareCheck,
+  },
+  array: {
+    title: "Array",
+    icon: faList.icon,
+  },
+  [types.dataType.object.dataTypeId]: {
+    title: "Property Object",
+    icon: faCube,
+  },
+};
+
+export const getDefaultData = (
+  typeId: string,
```

Comment:
Can this be `VersionedUri | "array"` to remove the need to cast?

Reply:
Done in 53b318c6cdd9de7817f0149e0ff3f0e439d2f91b. Thank you